ni/nimble
# Add locale support to number-text column

# Pull Request

## 🤨 Rationale

This is part of #1011 

This PR adds support for the number-text column using the `lang` token on the theme provider to format numbers in a locale-aware way.

## 👩‍💻 Implementation

I followed the same pattern used in the date-text column to listen for changes to the `lang` theme provider token and pass that into the formatters for the number-text column.

## 🧪 Testing

- Manually tested in storybook
- New unit test for the column that verifies the formatter is updated with the `lang` changes
- New unit tests for each formatter that verify the locale is used when formatting
    - Note: I updated the formatter tests to run all the test cases with an additional locale. On one hand this potentially seemed like "too many" locale tests, but I thought it was the best way to ensure that the locale was used in all internal formatters (e.g. the `default` formatter uses 3 formatters internally) and that we had coverage for a wide range of possible values to be formatted.

## ✅ Checklist
